tests/functional/aws-node-sdk/test/bucket/getCorsResponse.js

```diff
@@ -0,0 +1,994 @@
+import assert from 'assert';
+import http from 'http';
+import https from 'https';
+
+import { S3 } from 'aws-sdk';
+
+import conf from '../../../../../lib/Config';
+import getConfig from '../support/config';
+
+const config = getConfig('default', { signatureVersion: 'v4' });
+const s3 = new S3(config);
+
+const transport = conf.https ? https : http;
+const bucket = 'bucketcorstester';
+const hostname = process.env.AWS_ON_AIR ? 's3.amazonaws.com' :
+    'localhost';
+
+const port = process.env.AWS_ON_AIR ? 80 : 8000;
+
+const statusCode = {
+    200: 200,
+    NoSuchBucket: 404,
+    BadRequest: 400,
+    AccessForbidden: 403,
+};
+
+// AWS seems to take a bit long so sometimes by the time we send the request
+// the bucket has not yet been created or the bucket has been deleted.
+function _waitForAWS(callback, err) {
+    if (process.env.AWS_ON_AIR) {
+        setTimeout((done, err) => {
+            callback(err);
+        }, 5000);
+    } else {
+        callback(err);
+    }
+}
+
+function methodRequest(method, headers, code, headersResponse, done,
+objectKey) {
```

Comment:
Nicholas developed the tests first and I wanted to add some tests with object keys. I didn't want to add `undefined` before the callback to all the rest of the tests. But since it's convention to have the callback at the end, I decided to make the parameters a single object. https://github.com/scality/S3/pull/518/commits/8eebc1f9699dadad0eac10e1a07a57c8f7842b60